src/main/scala/co/adhoclabs/task/Serializer.scala

```diff
@@ -10,21 +10,24 @@ private[task] object Serializer {
   def serialize(obj: Any): Array[Byte] = {
     val b = new ByteArrayOutputStream()
     val o = new ObjectOutputStream(b)
-    o.writeObject(obj)
-    val result = b.toByteArray
-
-    o.close()
-    b.close()
-    result
+    try {
+      o.writeObject(obj)
+      o.flush()
+      b.toByteArray
+    } finally {
+      o.close()
+      b.close()
+    }
```

Comment:
does this run fine for you, it's not clear what's being returned there. is b.toByteArray being returned?